Pod/Classes/WPInputMediaPickerViewController.m

```diff
@@ -68,19 +67,69 @@ - (void)setupMediaPickerViewController {
 
 - (void)viewDidLayoutSubviews {
     [super viewDidLayoutSubviews];
+    [self configureCollectionView];
+}
 
-    CGFloat spacing = 1.0f;
-    CGFloat size = floorf((self.view.frame.size.height - spacing) / 2.0);
-    self.mediaPicker.options.cameraPreviewSize = CGSizeMake(1.5*size, 1.5*size);
+- (void)configureCollectionView {
+    CGFloat frameWidth = self.view.frame.size.width;
+    NSUInteger numberOfPhotosForLine = [self numberOfPhotosPerRow:frameWidth];
+    CGFloat photoSpacing = 1.0f;
+    CGFloat topInset = 5.0f;
+    CGFloat bottomInset = 10.0f;
+
+    CGFloat cellSize = [self.mediaPicker cellSizeForPhotosPerLineCount:numberOfPhotosForLine
+                                                          photoSpacing:photoSpacing
+                                                            frameWidth:frameWidth];
+
+    // Check the actual width of the content based on the computed cell size
+    // How many photos are we actually fitting per line?
+    CGFloat totalSpacing = (numberOfPhotosForLine - 1) * photoSpacing;
+    numberOfPhotosForLine = floorf((frameWidth - totalSpacing) / cellSize);
+
+    CGFloat contentWidth = (numberOfPhotosForLine * cellSize) + totalSpacing;
+
+    // If we have gaps in our layout, adjust to fit
+    if (contentWidth < frameWidth) {
+        cellSize = [self.mediaPicker cellSizeForPhotosPerLineCount:numberOfPhotosForLine
+                                                      photoSpacing:photoSpacing
+                                                        frameWidth:frameWidth];
+    }
+    
+    // Init and configure collection view layout
     UICollectionViewFlowLayout *layout = [[UICollectionViewFlowLayout alloc] init];
-    layout.scrollDirection = UICollectionViewScrollDirectionHorizontal;
-    layout.itemSize = CGSizeMake(size, size);
-    layout.minimumLineSpacing = spacing;
-    layout.minimumInteritemSpacing = spacing;
-    layout.sectionInset = UIEdgeInsetsMake(0, 5, 0, 10);
+    layout.scrollDirection = UICollectionViewScrollDirectionVertical;
```

Comment:
Could we have an option on the VC to allow horizontal or vertical scrolling? This will allow more flexibility for users of the library.